File: 02.每日一题/1021.删除最外层的括号/solution.py

class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        left_cnt = 0
        right_cnt = 0

        ans = ""

        for c in s:
            if c == ')':
                right_cnt += 1

            if left_cnt != right_cnt:
                ans += c

            if c == '(':
                left_cnt += 1

        return ans

File: 02.每日一题/1021.删除最外层的括号/solution.cpp
#include <string>

using namespace std;

class Solution {
 public:
  string removeOuterParentheses(string s) {
      string res;
      int left_cnt = 0, right_cnt = 0;

      for (char c : s) {
          if (c == ')') {
              right_cnt++;
          }

          if (left_cnt != right_cnt) {
              res.push_back(c);
          }

          if (c == '(') {
              left_cnt++;
          }
      }

      return res;
  }
};
